src/platform/pic32/uart.cpp

```diff
@@ -163,7 +181,32 @@ bool openxc::interface::uart::connected(UartDevice* device) {
             status = false;
             break;
     }
-    return status;
 
 #endif
+
+	// on apple devices, it takes some time for flow control to propagate to the device
```

Comment:
I believe so, yes. I monitored the UART lines with a scope during the process of connecting and sending data. Basically the "connected" indicator line on the RN42 goes high as soon as the devices connects to an iOS device. However, the application that handles the incoming iAP data is not necessary open at this time. If it is not, the flow control signal is supposed to go high (if we transmit too many bytes while there is no app open to handle our data, the link eventually breaks). However there appears to be a rather significant (and not consistent) delay in the flow control line propagating.

The current solution of a delay seems to be a simple and effective resolution given the current hardware and software setup, although it's inefficient. There is a more proper way I think, which involves an RN42 GPIO that goes high when an app has opened (which would replace the "connected" indicator line, although how this plays with non-iOS devices I'm not sure). There's also a set of diagnostic messages that can be enabled to indicate SESSION_OPEN and SESSION_CLOSED, but they are unsolicited so the RN42 control code needs to be able to handle that.